recipes/service.rb

```diff
@@ -74,17 +74,35 @@
   not_if { node['splunk']['server']['runasroot'] }
 end
 
-template '/etc/init.d/splunk' do
-  source 'splunk-init.erb'
-  mode 0700
-  variables(
-    splunkdir: splunk_dir,
-    runasroot: node['splunk']['server']['runasroot']
-  )
-end
+if node['init_package'] == 'systemd'
+  template '/usr/lib/systemd/system/splunk.service' do
+    source 'splunk-systemd.erb'
+    mode 0700
+    variables(
+      splunkdir: splunk_dir,
+      runasroot: node['splunk']['server']['runasroot']
+    )
+  end
+
+  service 'splunk' do
+    supports status: true, restart: true
+    provider Chef::Provider::Service::Systemd
+    action [:enable, :start]
+  end
+else
+  template '/etc/init.d/splunk' do
+    source 'splunk-init.erb'
+    mode 0700
+    variables(
+      splunkdir: splunk_dir,
+      runasroot: node['splunk']['server']['runasroot']
+    )
+    only_if { node['init_package'] == 'init' }
```

Comment:
Nope, that was a left-over from before I had added the if init_package code on line 77. I'll update my fork.